src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const obj = Object.assign({}, state);
+  const result = [];
+
+  actions.forEach(element => {
+    switch (element.type) {
+      case 'clear':
+        for (const key in obj) {
+          delete obj[key];
+        }
+        break;
+      case 'removeProperties':
+        element.keysToRemove.forEach(prop => {
+          delete obj[prop];
+        });
+        break;
+      default:
+        Object.assign(obj, element.extraData);
```

Comment:
use default case to handle errors. we shouldn't make assign if action type is invalid